internal/testsacc/alb_virtual_service_resource_test.go

```diff
@@ -0,0 +1,224 @@
+package testsacc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/helpers/testsacc"
+)
+
+var _ testsacc.TestACC = &ALBVirtualServiceResource{}
+
+const (
+	ALBVirtualServiceResourceName = testsacc.ResourceName("cloudavenue_alb_virtual_service")
+)
+
+type ALBVirtualServiceResource struct{}
+
+func NewALBVirtualServiceResourceTest() testsacc.TestACC {
+	return &ALBVirtualServiceResource{}
+}
+
+// GetResourceName returns the name of the resource.
+func (r *ALBVirtualServiceResource) GetResourceName() string {
+	return ALBVirtualServiceResourceName.String()
+}
+
+func (r *ALBVirtualServiceResource) DependenciesConfig() (resp testsacc.DependenciesConfigResponse) {
+	resp.Append(GetDataSourceConfig()[EdgeGatewayDataSourceName]().GetSpecificConfig("example_with_id"))
+	resp.Append(GetResourceConfig()[ALBPoolResourceName]().GetDefaultConfig)
+	// resp.Append(GetDataSourceConfig()[PublicIPResourceName]().GetDefaultConfig)
+	return
+}
+
+func (r *ALBVirtualServiceResource) Tests(ctx context.Context) map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test {
+	return map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test{
+		// TODO : Complete tests
+		// * First with service_type: HTTP and Simple Ports
+		"example": func(_ context.Context, resourceName string) testsacc.Test {
+			return testsacc.Test{
+				CommonChecks: []resource.TestCheckFunc{
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "name"),
+					resource.TestCheckResourceAttrSet(resourceName, "pool_id"),
+				},
+				// ! Create testing with service_type: HTTP
+				Create: testsacc.TFConfig{
+					TFConfig: testsacc.GenerateFromTemplate(resourceName, `
+					resource "cloudavenue_alb_virtual_service" "example" {
+						name = {{ generate . "name" }}
+						description = {{ generate . "description" }}
+						edge_gateway_id = data.cloudavenue_edgegateway.example_with_id.id
+						enabled = true
+						pool_id = cloudavenue_alb_pool.example.id
+						virtual_ip = "192.168.10.10"
+						service_type = "HTTP"
+						service_ports = [
+							{
+								port_start = 80
+							}
+						]
+					}`),
+					Checks: []resource.TestCheckFunc{
```

Comment:
Add tests : 

- Edgegateway name is Set
- ServiceEngineGroup is Set (The name of the serviceEngineGroup does it correspond to a format ?) 
- ServicePorts PortEnd/PortType/PortSSL

Reply:
I don't sea a clear format, i try: ^v[0-9]{3}w[0-9]{2}[e,i]pr[a-z]{12}[0-9]{2}$